[BUG] Don't suggest npm audit when it won't work (global installs)

### Is there an existing issue for this?

- [X] I have searched the existing issues

### This issue exists in the latest npm version

- [X] I am using the latest npm

### Current Behavior

Using `npm install --global <pack>` can suggest using `npm audit`. However, the latter fails and recommends creating a lock file. Creating a lock file is the wrong fix because the audit will be run for the local package, not for the global config.

### Expected Behavior

Either the `npm audit` recommendation should be removed if the lock file is not present, or there should be some other flag to `npm audit` which should be recommended instead to audit the global configuration.

### Steps To Reproduce

Environment:  (npm v8.1.4)

```
$ npm install --global verdaccio
# other output...
To address all issues (including breaking changes), run:
  npm audit fix --force

Run `npm audit` for details.
$ npm audit
npm ERR! code ENOLOCK
npm ERR! audit This command requires an existing lockfile.
npm ERR! audit Try creating one first with: npm i --package-lock-only
```

### Environment

- npm: npm 8.1.2
- Node.js: node v17.2.0
- npm config:
```ini
; node bin location = /Users/varun/.nvm/versions/node/v17.2.0/bin/node
; cwd = /Users/varun/Code/npm-testing
; HOME = /Users/varun
; Run `npm config ls -l` to show all defaults.
```

--- a/workspaces/arborist/lib/arborist/index.js
+++ b/workspaces/arborist/lib/arborist/index.js
@@ -77,7 +77,12 @@ class Arborist extends Base {
       replaceRegistryHost: options.replaceRegistryHost,
       lockfileVersion: lockfileVersion(options.lockfileVersion),
       installStrategy: options.global ? 'shallow' : (options.installStrategy ? options.installStrategy : 'hoisted'),
+      location: options.global ? 'global' : options.location,
     }
+    // don't audit when run in a non-project location
+    this.options.audit = (!this.options.location || this.options.location === 'project')
+      && options.audit !== false
+
     this.replaceRegistryHost = this.options.replaceRegistryHost =
       (!this.options.replaceRegistryHost || this.options.replaceRegistryHost === 'npmjs') ?
         'registry.npmjs.org' : this.options.replaceRegistryHost

--- a/workspaces/arborist/test/arborist/index.js
+++ b/workspaces/arborist/test/arborist/index.js
@@ -253,3 +253,13 @@ t.test('valid global/installStrategy values', t => {
   t.equal(new Arborist({ installStrategy: 'hoisted' }).options.installStrategy, 'hoisted')
   t.end()
 })
+
+t.test('disable audit when location is not project', t => {
+  t.equal(new Arborist({ location: 'global' }).options.audit, false)
+  t.equal(new Arborist({ location: undefined }).options.audit, true)
+  t.equal(new Arborist({ audit: undefined }).options.audit, true)
+  t.equal(new Arborist({ audit: false, location: 'project' }).options.audit, false)
+  t.equal(new Arborist({ global: true }).options.audit, false)
+  t.equal(new Arborist({ global: false }).options.audit, true)
+  t.end()
+})

